ethereum/core/src/main/java/tech/pegasys/teku/core/signatures/SlashingProtector.java

```diff
@@ -74,15 +88,35 @@ private ValidatorSigningRecord loadSigningRecord(final BLSPublicKey validator)
     record =
         dataAccessor
             .read(validatorRecordPath(validator))
-            .map(ValidatorSigningRecord::fromBytes)
+            .map(
+                val -> {
+                  try {
+                    return yamlProvider
+                        .getObjectMapper()
+                        .readValue(val.toArray(), SlashingProtectionRecord.class);
+                  } catch (IOException e) {
+                    LOG.trace(
+                        "error parsing slashing protection file " + validatorRecordPath(validator),
+                        e);
+                  }
+                  return new SlashingProtectionRecord();
+                })
+            .map(ValidatorSigningRecord::fromSlashingProtectionRecord)
```

Comment:
How did we manage to make this so much more complex?  Can we go back to how we were doing serialisation for `ValidatorSigningRecord`?  We seem to have wound up with yaml, IO and data model responsibilities scattered all through the code now.

Reply:
this is mostly because i had removed the separate yaml processor that had specific logic for how to export the validatorSigningRecord, and used the one in data. If we want two versions of yaml processor we could go back to the old version and clean that up a bit, but it was also doing things in a weird way... i could clean that up rather than pushing this through the other processor?

I guess i'd really like to have 1 yaml provider and 1 object definition for the serialized file, and its hard to access it from the import/export commands, so pushing it to one with minimal dependencies so that it could be reused was where i was headed...